src/nomad_simulations/general.py

```diff
@@ -192,8 +219,12 @@ def normalize(self, archive, logger) -> None:
         self.m_cache['system_ref'] = system_ref
 
         # Setting up the `branch_depth` in the parent-child tree
-        for system_parents in self.model_system:
-            system_parents.branch_depth = 0
-            if len(system_parents.model_system) == 0:
+        for system_parent in self.model_system:
+            system_parent.branch_depth = 0
+            if len(system_parent.model_system) == 0:
                 continue
-            self._set_system_branch_depth(system_parents)
+            self._set_system_branch_depth(system_parent)
+
+            if is_not_representative(system_parent, logger):
+                return
```

Comment:
Shouldn't this be `continue`? In case we have more normalization afterwards

Reply:
I was making a divide, so that all normalization below was for representative systems only, but maybe that's not idea, otherwise I can do:
```python
  if not is_not_representative(system_parent, logger):
      self.resolve_composition_formula(system_parent, logger)
```
I mean in that case I guess I would make a `is_representative` that returns the negative of the other function, but you get the point. Which do you prefer?

Reply:
I think `continue` is safer, it will simply go back to the for loop into the next element. With the current `return`, imagine that the first ModelSystem is not representative, but the next ones are, so this return will screw the resolve_composition_formula call for the systems that are actually representative.

Reply:
sorry, I somehow didn't realize this was in a loop, yes continue of course :facepalm: